exercises/clock/test/test_clock.c

```diff
@@ -16,406 +15,524 @@ void tearDown(void)
 }
 
 static void test_subtract_more_than_two_hours_with_borrow(void)
 {
    TEST_IGNORE();
-   int hour = 6;
-   int minute = 15;
-   int minute_offset = -160;
+   clock_t clock = { 6, 15 };
+   int minute_offset = 160;
    char *expected_clock_text = "03:35";
 
-   test_time_add_expected_value(expected_clock_text, hour, minute,
-                                minute_offset);
+   test_clock_subtract_expected_value(expected_clock_text, clock,
+                                      minute_offset);
 }
 
 static void test_subtract_more_than_one_day(void)
 {
    TEST_IGNORE();
-   int hour = 5;
-   int minute = 32;
-   int minute_offset = -1500;
+   clock_t clock = { 5, 32 };
+   int minute_offset = 1500;
    char *expected_clock_text = "04:32";
 
-   test_time_add_expected_value(expected_clock_text, hour, minute,
-                                minute_offset);
+   test_clock_subtract_expected_value(expected_clock_text, clock,
+                                      minute_offset);
 }
 
 static void test_subtract_more_than_two_days(void)
 {
    TEST_IGNORE();
-   int hour = 2;
-   int minute = 20;
-   int minute_offset = -3000;
+   clock_t clock = { 2, 20 };
+   int minute_offset = 3000;
    char *expected_clock_text = "00:20";
 
-   test_time_add_expected_value(expected_clock_text, hour, minute,
-                                minute_offset);
+   test_clock_subtract_expected_value(expected_clock_text, clock,
+                                      minute_offset);
+}
+
+static void test_compare_clocks_with_same_time(void)
+{
+   TEST_IGNORE();
+   clock_t clock1 = { 15, 37 };
+   clock_t clock2 = { 15, 37 };
+
+   TEST_ASSERT_TRUE(clock_is_equal(clock1, clock2));
+}
+
+static void test_compare_clocks_a_minute_apart(void)
+{
+   TEST_IGNORE();
+   clock_t clock1 = { 15, 36 };
+   clock_t clock2 = { 15, 37 };
+
+   TEST_ASSERT_FALSE(clock_is_equal(clock1, clock2));
+}
+
+static void test_compare_clocks_an_hour_apart(void)
+{
+   TEST_IGNORE();
+   clock_t clock1 = { 14, 37 };
+   clock_t clock2 = { 15, 37 };
+
+   TEST_ASSERT_FALSE(clock_is_equal(clock1, clock2));
+}
+
+static void test_compare_clocks_with_hour_overflow(void)
+{
+   TEST_IGNORE();
+   clock_t clock1 = { 10, 37 };
+   clock_t clock2 = { 34, 37 };
+
+   TEST_ASSERT_TRUE(clock_is_equal(clock1, clock2));
+}
+
+static void test_compare_clocks_with_hour_overflow_by_several_days(void)
+{
+   TEST_IGNORE();
+   clock_t clock1 = { 3, 11 };
+   clock_t clock2 = { 99, 11 };
+
+   TEST_ASSERT_TRUE(clock_is_equal(clock1, clock2));
+}
+
+static void test_compare_clocks_with_negative_hour(void)
+{
+   TEST_IGNORE();
+   clock_t clock1 = { 22, 40 };
+   clock_t clock2 = { -2, 40 };
+
+   TEST_ASSERT_TRUE(clock_is_equal(clock1, clock2));
+}
+
+static void test_compare_clocks_with_negative_hour_that_wraps(void)
+{
+   TEST_IGNORE();
+   clock_t clock1 = { 17, 3 };
+   clock_t clock2 = { -31, 3 };
+
+   TEST_ASSERT_TRUE(clock_is_equal(clock1, clock2));
+}
+
+static void
```

Comment:
I kind of hate our formatter

Reply:
The 3 space indent catches me out every time 😖 
I know it's the Linux style but it is idiosyncratic in some parts.
Scope to adjust to more generally accepted formatting? (e.g. 4 space indent, 120 char line, etc)

Reply:
Oh, for my intital `clock_t clock = { .hour = 12, .minute = 5 };` it did `clock_t clock = {.hour = 12,.minute = 5 };`  🙅‍♂

Reply:
I'm mostly just complaining to complain. It's hard to find a formatter/configuration that doesn't have some sort of odd wart.